web/src/lib.rs

```diff
@@ -14,16 +14,16 @@ mod router;
 
 pub async fn init_server(app_state: AppState) -> Result<()> {
     // These will probably come from app_state.config (command line)
-    let host = "127.0.0.1";
-    let port = 3000;
+    let host = app_state.config.interface.as_ref().unwrap();
```

Comment:
Curious why we need the as_ref here?

Reply:
The unwrap() consumes the value otherwise and rustc complains about the ownership move farther down in the code on line 27.